From 5f4e7262adc12b5227a392ca78df8968d5435b27 Mon Sep 17 00:00:00 2001
From: Cristian Vijelie <cristianvijelie@gmail.com>
Date: Fri, 14 Aug 2020 13:05:57 +0300
Subject: [UNIKRAFT/LIBNETTLE 3/4] minor changes to Makefile.uk

---
 Makefile.uk | 27 +++++----------------------
 1 file changed, 5 insertions(+), 22 deletions(-)

diff --git a/Makefile.uk b/Makefile.uk
index 6e896da..460a078 100644
--- a/Makefile.uk
+++ b/Makefile.uk
@@ -1,15 +1,16 @@
 $(eval $(call addlib_s,libnettle,$(CONFIG_LIBNETTLE)))
 
 LIBNETTLE_VERSION=3.6
-LIBNETTLE_TARNAME=nettle-$(LIBNETTLE_VERSION)
-LIBNETTLE_URL=https://ftp.gnu.org/gnu/nettle/$(LIBNETTLE_TARNAME).tar.gz
+LIBNETTLE_ZIPNAME=nettle-$(LIBNETTLE_VERSION)
+LIBNETTLE_URL=https://ftp.gnu.org/gnu/nettle/$(LIBNETTLE_ZIPNAME).tar.gz
 $(eval $(call fetch,libnettle,$(LIBNETTLE_URL)))
 
 LIBNETTLE_PATCHDIR=$(LIBNETTLE_BASE)/patches
+$(eval $(call patch,libnettle,$(LIBNETTLE_PATCHDIR),$(LIBNETTLE_ZIPNAME)))
 
-LIBNETTLE_SUBDIR = $(LIBNETTLE_ORIGIN)/nettle-$(LIBNETTLE_VERSION)
+LIBNETTLE_SUBDIR = $(LIBNETTLE_ORIGIN)/$(LIBNETTLE_ZIPNAME)
 
-LIBNETTLE_CFLAGS += -I. -I$(LIBNETTLE_BASE)/include -DHAVE_CONFIG_H -g -O2 -ggdb3 -Wall -W -Wno-sign-compare   -Wmissing-prototypes -Wmissing-declarations -Wstrict-prototypes   -Wpointer-arith -Wbad-function-cast -Wnested-externs -fpic -MT $@ -MD -MP -MF $@.d
+LIBNETTLE_CFLAGS = -I. -I$(LIBNETTLE_BASE)/include -DHAVE_CONFIG_H -g -O2 -ggdb3 -Wall -W -Wno-sign-compare   -Wmissing-prototypes -Wmissing-declarations -Wstrict-prototypes   -Wpointer-arith -Wbad-function-cast -Wnested-externs -fpic -MT $@ -MD -MP -MF $@.d
 
 LIBNETTLE_SRCS-y += $(LIBNETTLE_SUBDIR)/aes-decrypt-internal.s
 LIBNETTLE_SRCS-y += $(LIBNETTLE_SUBDIR)/aes-decrypt.c
@@ -208,24 +209,6 @@ LIBNETTLE_SRCS-y += $(LIBNETTLE_SUBDIR)/yarrow_key_event.c
 LIBNETTLE_SRCS-y += $(LIBNETTLE_SUBDIR)/xts.c
 LIBNETTLE_SRCS-y += $(LIBNETTLE_SUBDIR)/xts-aes128.c
 LIBNETTLE_SRCS-y += $(LIBNETTLE_SUBDIR)/xts-aes256.c
-# LIBNETTLE_SRCS-y += $(LIBNETTLE_SUBDIR)/aes-decrypt-internal.s
-# LIBNETTLE_SRCS-y += $(LIBNETTLE_SUBDIR)/aes-encrypt-internal.s
-# LIBNETTLE_SRCS-y += $(LIBNETTLE_SUBDIR)/camellia-crypt-internal.s
-# LIBNETTLE_SRCS-y += $(LIBNETTLE_SUBDIR)/chacha-core-internal.s
-# LIBNETTLE_SRCS-y += $(LIBNETTLE_SUBDIR)/md5-compress.s
-# LIBNETTLE_SRCS-y += $(LIBNETTLE_SUBDIR)/memxor.s
-# LIBNETTLE_SRCS-y += $(LIBNETTLE_SUBDIR)/memxor3.s
-# LIBNETTLE_SRCS-y += $(LIBNETTLE_SUBDIR)/poly1305-internal.s
-# LIBNETTLE_SRCS-y += $(LIBNETTLE_SUBDIR)/salsa20-core-internal.s
-# LIBNETTLE_SRCS-y += $(LIBNETTLE_SUBDIR)/salsa20-crypt.s 
-# LIBNETTLE_SRCS-y += $(LIBNETTLE_SUBDIR)/sha1-compress.s
-# LIBNETTLE_SRCS-y += $(LIBNETTLE_SUBDIR)/sha256-compress.s
-# LIBNETTLE_SRCS-y += $(LIBNETTLE_SUBDIR)/sha512-compress.s
-# LIBNETTLE_SRCS-y += $(LIBNETTLE_SUBDIR)/sha3-permute.s
-# LIBNETTLE_SRCS-y += $(LIBNETTLE_SUBDIR)/serpent-encrypt.s
-# LIBNETTLE_SRCS-y += $(LIBNETTLE_SUBDIR)/serpent-decrypt.s
-# LIBNETTLE_SRCS-y += $(LIBNETTLE_SUBDIR)/umac-nh.s
-# LIBNETTLE_SRCS-y += $(LIBNETTLE_SUBDIR)/umac-nh-n.s
 LIBNETTLE_SRCS-y += $(LIBNETTLE_SUBDIR)/gcm-hash8.s
 
 # Run ./configure
-- 
2.25.1

